### PR TITLE
always typecheck the rhs of let expressions

### PR DESCRIPTION
**Pull request by nickchapman-da**
Whilst setting up the stack-safety tests for type-checking, I noticed what looked like a bug in the typechecking code (`typeOfLet`) for `ELet` expressions. If the `Binding` was un-named, the `rhs` was not traversed, and so any type errors it might contain were not discovered.

This PR:
- fixes the bug in `Typing.scala`
- extends `ExpParser` to allow the `Binding(None,...)` AST form to be constructed (needed for the test)
- adds a test in `TypingSpec.scala` (which would fail without the corresponding fix)

### CODE REVIEW

**Review by remyhaemmerle-da — APPROVED** [review]
Good catch. 
Thanks.

File: daml-lf/parser/src/main/scala/com/digitalasset/daml/lf/testing/parser/ExprParser.scala [hunk 1]
```diff
@@ -188,7 +188,10 @@ private[parser] class ExprParser[P](parserParameters: ParserParameters[P]) {
   private def binding(sep: Token): Parser[Binding] =
     id ~ (`:` ~> typ) ~ (sep ~> expr) ^^ { case vName ~ t ~ value =>
       Binding(Some(vName), t, value)
-    }
+    } |
+      `_` ~> (`:` ~> typ) ~ (sep ~> expr) ^^ { case t ~ value =>
+        Binding(None, t, value)
+      }
```
Comment (remyhaemmerle-da) [inline daml-lf/parser/src/main/scala/com/digitalasset/daml/lf/testing/parser/ExprParser.scala:194]: Could you add a test in `ParserSpec` ?

Comment (nickchapman-da) [reply daml-lf/parser/src/main/scala/com/digitalasset/daml/lf/testing/parser/ExprParser.scala:194]: done

**Pull request merged**